data/integration_tests/percolator/IntegrationTest_Percolator_Speed.py

```diff
@@ -0,0 +1,242 @@
+    print("Await PIDs:\t\t" + str(args.processes))
+    print("Await other tests?\t" + str(args.await_tests))
+    print("Number of iterations:\t" + str(args.runs))
+    print("Comments:\t\t" + str(args.comments))
+    print("Testing command:\t" + getTerminalCommand(args.flags, args.data))
+
+def getRunTimes(args):
+    runTimes = []
+    for i in range(1,args.runs+1):
+        terminalCommand = getTerminalCommand(args.flags,args.data)
+        terminalCommand = terminalCommand.split(' ')
+        terminalCommand = list(filter(None, terminalCommand))
+        p = subprocess.Popen(terminalCommand, stderr=PIPE , stdout=subprocess.DEVNULL , text=True)
+        print("Running subprocess: " + str(p.pid))
+        p.wait()
+        removeTemporaryFile()
+        runTime = getRunTime(p.stderr.read())
+        runTimes.append(runTime)
+    return runTimes
+
+def printResults(runTimes):
+    print(lineSeparatingText)
+    numElements = len(runTimes)
+    totalCPUSec = 0
+    totalWallSec = 0
+    minCPUSec = 99999999999
+    minWallSec = 99999999999
+    maxCPUSec = -99999999999
+    maxWallSec = -99999999999
+    print("Execution time in CPU and wall clock seconds, precision set to 3 decimals.")
+
+    for i in range(0,numElements):
+        cpuSec = runTimes[i][0]
+        wallSec = runTimes[i][1]
+        totalCPUSec += cpuSec
+        totalWallSec += wallSec
+        minCPUSec = min(minCPUSec, cpuSec)
+        minWallSec = min(minWallSec, wallSec)
+        maxCPUSec = max(maxCPUSec, cpuSec)
+        maxWallSec = max(maxWallSec, wallSec)
+        print("Run " + str(i+1) + "\tCPU: " + '{0:.3f}'.format(cpuSec) + " " + "Wall: " + str(int(wallSec)) )
+    print(lineSeparatingText)
+    print("Max\tCPU: " + '{0:.3f}'.format(maxCPUSec) + " " + "Wall: " + str(int(maxWallSec)))
+    print("Min\tCPU: " + '{0:.3f}'.format(minCPUSec) + " " + "Wall: " + str(int(minWallSec)))
+    print("Mean\tCPU: " + '{0:.3f}'.format(totalCPUSec/numElements) + " "  + "Wall: " + str(Decimal('{0:.3f}'.format(totalWallSec/numElements)).normalize()) )
+
+def getDataFromUrl(args):
+    global success
+    if (len(args.data) == 0 and checkIfTestDataExist() is False):
+        try:
+            downloadTestData()
+        except HTTPError as e:
+            print("Failed to download test-data. HTTP Error code: ", e.code)
+            success = False
+            return
+        except URLError as e:
+            print("Failed to download test-data. Reason: ", e.reason)
+            success = False
+            return
+        except:
+            print("Failed to download test-data.")
+            success = False
+            return
+            
+        print("Extracting...")
+        extractTarFile(testDataFolderPath + "/" + downloadedDataFileName)
+        print("Done extracting.")
+    if(len(args.data) == 0):
+        args.data = testDataFullPath
+
+
+args = getArguments()
+if(args.await_tests is True or args.processes is not None):
+    import psutil
```

Comment:
psutil is needed to check if PIDs contain strings in a cross-platform way. os-commands may not always work for Windows. I Will add a comment.

Reply:
I feel like using the pid returned by subprocess.Popen() directly, instead of searching the pid table for strings, would have been more straightforward, and avoided the need for psutil's functionality. But if the dependency on psutil is not an issue for other users, then it's largely an academic point.